File: Models/Shift.cs
using dotnet_backend_freshmart.Models.Enums;

namespace dotnet_backend_freshmart.Models
{
    public class Shift
    {
        public Guid Id { get; set; }

        public Guid EmployeeId { get; set; }

        // Khóa ngoại liên kết tới Nhân viên thu ngân (Employee)
        public virtual Employee? Employee { get; set; }

        public string ShiftName { get; set; } = string.Empty;

        public DateTime StartTime { get; set; } = DateTime.UtcNow;

        public DateTime? EndTime { get; set; }

        public decimal StartingCash { get; set; } = 0;

        public decimal ExpectedCash { get; set; } = 0;

        public decimal? ActualCash { get; set; }

        public decimal TotalRevenue { get; set; } = 0;

        public int OrderCount { get; set; } = 0;

        public ShiftStatus Status { get; set; } = ShiftStatus.Active;

        public string? Notes { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    }
}

File: freshmart/src/types/shift.ts
export type ShiftStatus = 'Active' | 'Closed' | 'active' | 'closed';

export interface Shift {
  id: string;
  employeeId: string;
  cashierName: string;
  cashierCode: string;
  shiftName: string;
  startTime: string;
  endTime: string | null;
  startingCash: number;
  expectedCash: number;
  actualCash: number | null;
  difference: number | null;
  totalRevenue: number;
  orderCount: number;
  status: ShiftStatus;
  notes: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface OpenShiftRequest {
  employeeId?: string;
  shiftName: string;
  startingCash: number;
  notes?: string;
}

export interface CloseShiftRequest {
  actualCash: number;
  notes?: string;
}

export interface ShiftReport {
  shiftId: string;
  shiftName: string;
  cashierName: string;
  cashierCode: string;
  startTime: string;
  endTime: string | null;
  status: ShiftStatus;
  startingCash: number;
  cashSales: number;
  vietQRSales: number;
  cardSales: number;
  totalRevenue: number;
  expectedCash: number;
  actualCash: number | null;
  cashDifference: number | null;
  totalOrders: number;
  averageOrderValue: number;
  notes: string | null;
}

export interface ShiftFilterParams {
  employeeId?: string;
  fromDate?: string;
  toDate?: string;
}
